runDocker.ts

```diff
@@ -78,9 +79,25 @@ export async function removeContainer(containerId: string) {
   await docker.getContainer(containerId).remove()
 }
 
+async function pullImage(imageName: string) {
+  try {
+    const pullStream = await docker.pull(imageName)
+    pullStream.pipe(process.stdout)
+    await new Promise((resolve, reject) => {
+      pullStream.on('end', resolve)
+      pullStream.on('error', reject)
+    })
+    console.log(`Image ${imageName} pulled successfully`)
+  } catch (error) {
+    console.error(`Error pulling image ${imageName}:`, error)
+  }
+}
```

Comment:
This isn't the recommended pattern for waiting for the pull to finish. See https://github.com/apocas/dockerode?tab=readme-ov-file#helper-functions.

Also, note that this PR is titled "Remove defined credentials". The Docker pull has nothing to do with that. Please move this to a separate PR.